src/extension/src/scripts/run-tests.ts

```diff
@@ -0,0 +1,150 @@
+let child_process = require("child_process");
+const path = require("path");
+const fs = require("fs");
+
+const testFolder = path.join(__dirname, "..", "..", "src", "template_test");
+console.log(testFolder);
+let files: string[] = [];
+
+//Recursively delete all generated projects in template_test
+let deleteFolderRecursive = function(pathname: string) {
+  let files = [];
+  if (fs.existsSync(pathname)) {
+    files = fs.readdirSync(pathname);
+    files.forEach(function(file: string, index: number) {
+      var curPath = path.join(pathname, file);
+      if (fs.lstatSync(curPath).isDirectory()) {
+        deleteFolderRecursive(curPath);
+      } else {
+        // delete file
+        fs.unlinkSync(curPath);
+      }
+    });
+    fs.rmdirSync(pathname);
+    console.log(`deleted ${pathname}`);
+  }
+};
+
+fs.readdirSync(testFolder).forEach((file: string) => {
+  files.push(file.toString());
+});
+
+function kill(pid: any) {
+  try {
+    child_process.execSync("taskkill /PID " + pid + " /T /F", function(
+      error: any,
+      stdout: any,
+      stderr: any
+    ) {
+      console.log("Stdout from kill: " + stdout);
+      console.log("Stderr from kill:" + stderr);
+      if (error !== null) {
+        console.log("Error from kill: " + error);
+      }
+    });
+  } catch (err) {
+    console.log(`Error from taskkill: ${err}`);
+  }
+}
+
+function sleep(ms: number) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+async function asyncForEach(array: string[], callback: any) {
+  for (let index = 0; index < array.length; index++) {
+    await callback(array[index], index, array);
+  }
+}
+
+asyncForEach(files, async (file: string) => {
+  console.log(`Current file: ${file}`);
+
+  const currDir = path.join(testFolder, file, file);
+  child_process.execSync("yarn install", {
+    cwd: currDir,
+    stdio: "inherit"
+  });
+  let serverProcess;
+  let testProcess;
+  try {
+    serverProcess = child_process.exec(
+      "yarn start",
+      {
+        cwd: path.join(testFolder, file, file),
+        stdio: "inherit"
+      },
+      (error: any, stdout: any, stderr: any) => {
+        if (error) {
+          console.error(`Error from running yarn start: ${error}`); //TODO: mark failure
```

Comment:
is there more work for this TODO in this PR?

Reply:
Should we throw here?

Reply:
Oh I just forgot to remove this haha - it's gone now